elm-street.cabal

```diff
@@ -109,8 +109,8 @@ executable run-backend
   main-is:             Main.hs
   other-modules:       Api
 
-  build-depends:       servant >= 0.14 && < 0.19
-                     , servant-server >= 0.14 && < 0.19
+  build-depends:       servant >= 0.14 && < 0.20
+                     , servant-server >= 0.14 && < 0.20
```

Comment:
I kind of find this version management contra-productive. I think we constrain these boundaries too much.

Reply:
https://www.parsonsmatt.org/2020/05/07/on_pvp_restrictive_bounds.html

Reply:
Would you rather use 1.0 as an exclusive upper bound, or remove it entirely? I have to admit that having our project fails to build because of an unfortunate constraint on a dependency we don't even use (elm-street example project) is annoying.

Reply:
Thank you @jhrcek for this very interesting (and long) blog post 😀
Though I'm still a little fuzzy on what might be the best move for us right now. I understand the final part and can try to see how I can configure cabal to ignore the elm-street example (more specifically ignore the version constraint on servant) and leave the library alone. But because we now know that elm-street's example works well with servant 0.19, we might as well update it and spare some headaches to the library's potential users, shouldn't we? If we decide to change the upper bound, if I've understood the argument correctly, we can only assume correct use of the PVP and use 0.19, is that right?

Reply:
I just posted the article to say that there might be good reason to keep restrictive version bounds in some case but this particular case of an example application is not one of them.
I'd say let's do the easiest thing for us and remove the upper bound.
The worst thing that can happen is that the example will break in the future.

Reply:
There are always tradeoffs. I kind of waited for seeing PR like this for project like this one to bring this up since this seems like a low impact place where we could consider changing the approach...

As for my opinion I think these bounds are useless in any app that uses package sets like stackage (via stack) or nixpkgs. There might be some reasons to have these for libraries that want to provide some hints for standard cabal version resolver but frankly I think my personal libraries kind off [speak for themself - so yeah I'm kind of biased against these bounds](https://github.com/turboMaCk/aeson-combinators/blob/master/aeson-combinators.cabal#L37-L51)

Reply:
as for my justification - I'm really a simple man. If I don't know I resolve to counting occurrences where some approach helped and when it hurt. And to me it seems that upper bounds most often hurt rather than help. There are far more PRs and people having issues with too restrictive upper bounds than there are instances where it's hard to build a project because of too relaxed upper bound.

Reply:
Sounds good to me! The upper bound has been removed entirely.